src/app/features/container/containers/home/home-header.tsx

```diff
@@ -0,0 +1,61 @@
+import { ReactNode, useState } from 'react';
+
+import { SettingsSelectors } from '@tests/selectors/settings.selectors';
+import { Flex, Grid, GridItem, HStack, styled } from 'leather-styles/jsx';
+
+import { HamburgerIcon } from '@leather.io/ui';
+
+import { SwitchAccountDialog } from '@app/features/dialogs/switch-account-dialog/switch-account-dialog';
+
+import { Settings } from '../../../settings/settings';
+
+interface HomeHeaderProps {
+  networkBadge?: ReactNode;
+  logo?: ReactNode;
+}
+
+export function HomeHeader({ networkBadge, logo }: HomeHeaderProps) {
+  const [isShowingSwitchAccount, setIsShowingSwitchAccount] = useState(false);
+  return (
+    <>
+      {isShowingSwitchAccount && (
```

Comment:
`isShowingSwitchAccount` was located at the top container but when I was testing other re-rendering issues I noticed that when we open/close the dialog it was re-rendering the whole app. 

I decided to localise instances of the `SwitchAccountDialog` to where they are called from. This means some duplicate code but helps to avoid re-rendering the app un-necessarily.